### PR TITLE
Compatibility with the seimgroup (<>) operator from Prelude & minor export change

### PR DESCRIPTION
**Pull request by osense**

### CODE REVIEW

File: src/Database/HaskellDB/Query.hs [hunk 1]
```diff
@@ -37,7 +37,7 @@ module Database.HaskellDB.Query (
 	     , isNull, notNull
 	     , fromNull, fromVal
 	     , constant, constVal, constNull, constExpr
-	     , param, namedParam, Args, func, cast
+	     , param, namedParam, Args(..), func, cast
```
Comment (m4dc4p) [inline src/Database/HaskellDB/Query.hs]: Why export `_args`? It's basically an implementation detail.

**Comment by osense** [comment]
We define our own instance of `Args` for use with the PSQL function `date_part`:
```haskell
instance Args (String -> Expr a -> Expr b) where
  arg_ n@"date_part" [] =
      \p (Expr d) -> Expr $ FunExpr n [ConstExpr $ StringLit p,d]
  arg_ n _ = error $ "Invalid function name: " ++ n

date_part :: String -> Expr (Maybe CalendarTime) -> Expr Integer
date_part = func "date_part"
```
Then, we can use it like so in projections:
```haskell
  project
    $ F.id << event!F.id
    # F.year << date_part "year" (event!F.startDate)
```
There might be a better way to achieve this, I'm not sure. I'm not terribly fluent with haskelldb, but I was migrating a project from GHC 6.12 to 8.4.4 and it seems at some point `arg_` became private, thus breaking our code.

**Comment by m4dc4p** [comment]
You only need to use the `func` function to define custom functions - it takes any number of arguments - they just have to be convertable to `Expr`. 

`date_part` is actually an example in the `func` docs: https://github.com/m4dc4p/haskelldb/blob/master/src/Database/HaskellDB/Query.hs#L565.

However, I'm glad to merge the `<>` changes, if you'd like to update this PR.

**Comment by osense** [comment]
Alright I see, thanks! I removed the commit.

**Pull request merged**